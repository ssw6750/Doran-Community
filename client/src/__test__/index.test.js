// __tests__/index.test.js
import { render, screen } from "@testing-library/react";
import Home from "../pages/index";

describe("Home", () => {
  it("renders a heading", () => {
    const { container } = render(<Home />);

    const heading = screen.getByText("인기 게시판");
    
    expect(heading).toBeInTheDocument();

    // expect(container).toMatchSnapshot();
  });
});
