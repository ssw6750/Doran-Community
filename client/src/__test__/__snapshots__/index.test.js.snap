// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders a heading 1`] = `
<div>
  <div
    class="max-w-5xl px-4 pt-2 mx-auto"
  >
    <div>
      <div
        class="hidden md:block"
      >
        <div
          class="justify-content flex mb-2"
        >
          <button
            class="h-36 bg-gray-500 flex-auto first:ml-0 mx-1 last:mr-0 rounded relative"
            style="background-repeat: no-repeat; background-size: cover; background-position: center;"
          >
            <div
              class="absolute text-white bottom-0 left-0 right-0 m-2"
            >
              <div
                class="truncate"
              >
                [라이엇 스토어] 2023 VCT 마스터스 컬렉션
              </div>
              <div
                class="truncate w-inherit text-xs"
              >
                2023 VCT 마스터스 도쿄 상품 컬렉션으로 마스터스 도쿄 대회를 기념하세요!
              </div>
            </div>
          </button>
          <button
            class="h-36 bg-gray-500 flex-auto first:ml-0 mx-1 last:mr-0 rounded relative"
            style="background-repeat: no-repeat; background-size: cover; background-position: center;"
          >
            <div
              class="absolute text-white bottom-0 left-0 right-0 m-2"
            >
              <div
                class="truncate"
              >
                발로란트 필수 고지사항 및 VP 정책 변경 안내
              </div>
              <div
                class="truncate w-inherit text-xs"
              >
                발로란트 VP 정책 및 계약 관련 필수 고지 사항이 변경될 예정입니다.
              </div>
            </div>
          </button>
          <button
            class="h-36 bg-gray-500 flex-auto first:ml-0 mx-1 last:mr-0 rounded relative"
            style="background-repeat: no-repeat; background-size: cover; background-position: center;"
          >
            <div
              class="absolute text-white bottom-0 left-0 right-0 m-2"
            >
              <div
                class="truncate"
              >
                에피소드 7 업데이트 사전예약 안내
              </div>
              <div
                class="truncate w-inherit text-xs"
              >
                발로란트의 새로운 시작, 에피소드 7 업데이트에 사전예약하고 특별한 보상도 획득하세요!
              </div>
            </div>
          </button>
          <button
            class="h-36 bg-gray-500 flex-auto first:ml-0 mx-1 last:mr-0 rounded relative"
            style="background-repeat: no-repeat; background-size: cover; background-position: center;"
          >
            <div
              class="absolute text-white bottom-0 left-0 right-0 m-2"
            >
              <div
                class="truncate"
              >
                2023년 6월 8일 (목) 서버 점검 안내 (완료)
              </div>
              <div
                class="truncate w-inherit text-xs"
              >
                2023 VCT 마스터스 도쿄 상품 컬렉션으로 마스터스 도쿄 대회를 기념하세요!
              </div>
            </div>
          </button>
        </div>
      </div>
      <div
        class="overflow-hidden w-inherit relative border mb-2 md:hidden hover:cursor-pointer"
      >
        <div
          class="flex justify-content w-[400%] -ml-[0%] transition-all "
        >
          <div
            class="absolute z-10 top-[50%] translate-y-[-50%] left-2 bg-gray-500 opacity-50 rounded-full text-white"
          >
            <svg
              class="text-4xl"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M217.9 256L345 129c9.4-9.4 9.4-24.6 0-33.9-9.4-9.4-24.6-9.3-34 0L167 239c-9.1 9.1-9.3 23.7-.7 33.1L310.9 417c4.7 4.7 10.9 7 17 7s12.3-2.3 17-7c9.4-9.4 9.4-24.6 0-33.9L217.9 256z"
              />
            </svg>
          </div>
          <div
            class="absolute z-10 top-[50%] translate-y-[-50%] right-2 bg-gray-500 opacity-50 rounded-full text-white"
          >
            <svg
              class="text-4xl"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M294.1 256L167 129c-9.4-9.4-9.4-24.6 0-33.9s24.6-9.3 34 0L345 239c9.1 9.1 9.3 23.7.7 33.1L201.1 417c-4.7 4.7-10.9 7-17 7s-12.3-2.3-17-7c-9.4-9.4-9.4-24.6 0-33.9l127-127.1z"
              />
            </svg>
          </div>
          <div
            class="h-36 bg-gray-500 rounded relative flex-none w-[25%]"
            style="background-repeat: no-repeat; background-size: cover; background-position: center;"
          >
            <div
              class="absolute text-white bottom-0 left-0 right-0 m-2"
            >
              <div
                class="truncate"
              >
                [라이엇 스토어] 2023 VCT 마스터스 컬렉션
              </div>
              <div
                class="truncate w-inherit text-xs"
              >
                2023 VCT 마스터스 도쿄 상품 컬렉션으로 마스터스 도쿄 대회를 기념하세요!
              </div>
            </div>
          </div>
          <div
            class="h-36 bg-gray-500 rounded relative flex-none w-[25%]"
            style="background-repeat: no-repeat; background-size: cover; background-position: center;"
          >
            <div
              class="absolute text-white bottom-0 left-0 right-0 m-2"
            >
              <div
                class="truncate"
              >
                발로란트 필수 고지사항 및 VP 정책 변경 안내
              </div>
              <div
                class="truncate w-inherit text-xs"
              >
                발로란트 VP 정책 및 계약 관련 필수 고지 사항이 변경될 예정입니다.
              </div>
            </div>
          </div>
          <div
            class="h-36 bg-gray-500 rounded relative flex-none w-[25%]"
            style="background-repeat: no-repeat; background-size: cover; background-position: center;"
          >
            <div
              class="absolute text-white bottom-0 left-0 right-0 m-2"
            >
              <div
                class="truncate"
              >
                에피소드 7 업데이트 사전예약 안내
              </div>
              <div
                class="truncate w-inherit text-xs"
              >
                발로란트의 새로운 시작, 에피소드 7 업데이트에 사전예약하고 특별한 보상도 획득하세요!
              </div>
            </div>
          </div>
          <div
            class="h-36 bg-gray-500 rounded relative flex-none w-[25%]"
            style="background-repeat: no-repeat; background-size: cover; background-position: center;"
          >
            <div
              class="absolute text-white bottom-0 left-0 right-0 m-2"
            >
              <div
                class="truncate"
              >
                2023년 6월 8일 (목) 서버 점검 안내 (완료)
              </div>
              <div
                class="truncate w-inherit text-xs"
              >
                2023 VCT 마스터스 도쿄 상품 컬렉션으로 마스터스 도쿄 대회를 기념하세요!
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="flex"
    >
      <div
        class="w-full md:mr-3 md:w-8/12"
      >
        <div
          class="flex p-4 items-end w-full h-24 border border-basic-gray-third bg-white mb-2 relative gap-6 text-gray-400"
        >
          <p
            class="absolute top-4 text-lg text-basic-black-second font-ptBlack"
          >
            전체
          </p>
          <label
            class="flex justify-center items-center hover:text-basic-red text-sm "
          >
            <input
              class="appearance-none"
              type="radio"
              value="latest"
            />
            <svg
              class="text-lg"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 160c-52.9 0-96 43.1-96 96s43.1 96 96 96 96-43.1 96-96-43.1-96-96-96zm246.4 80.5l-94.7-47.3 33.5-100.4c4.5-13.6-8.4-26.5-21.9-21.9l-100.4 33.5-47.4-94.8c-6.4-12.8-24.6-12.8-31 0l-47.3 94.7L92.7 70.8c-13.6-4.5-26.5 8.4-21.9 21.9l33.5 100.4-94.7 47.4c-12.8 6.4-12.8 24.6 0 31l94.7 47.3-33.5 100.5c-4.5 13.6 8.4 26.5 21.9 21.9l100.4-33.5 47.3 94.7c6.4 12.8 24.6 12.8 31 0l47.3-94.7 100.4 33.5c13.6 4.5 26.5-8.4 21.9-21.9l-33.5-100.4 94.7-47.3c13-6.5 13-24.7.2-31.1zm-155.9 106c-49.9 49.9-131.1 49.9-181 0-49.9-49.9-49.9-131.1 0-181 49.9-49.9 131.1-49.9 181 0 49.9 49.9 49.9 131.1 0 181z"
              />
            </svg>
            <span
              class="ml-2"
            >
              최신
            </span>
          </label>
          <label
            class="flex justify-center items-center hover:text-basic-red text-sm text-basic-red"
          >
            <input
              checked=""
              class="appearance-none"
              type="radio"
              value="popularity"
            />
            <svg
              class="text-lg"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 384 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M216 23.86c0-23.8-30.65-32.77-44.15-13.04C48 191.85 224 200 224 288c0 35.63-29.11 64.46-64.85 63.99-35.17-.45-63.15-29.77-63.15-64.94v-85.51c0-21.7-26.47-32.23-41.43-16.5C27.8 213.16 0 261.33 0 320c0 105.87 86.13 192 192 192s192-86.13 192-192c0-170.29-168-193-168-296.14z"
              />
            </svg>
            <span
              class="ml-2"
            >
              인기
            </span>
          </label>
          <label
            class="flex justify-center items-center hover:text-basic-red text-sm "
          >
            <input
              class="appearance-none"
              type="radio"
              value="recommendation"
            />
            <svg
              class="text-lg"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M104 224H24c-13.255 0-24 10.745-24 24v240c0 13.255 10.745 24 24 24h80c13.255 0 24-10.745 24-24V248c0-13.255-10.745-24-24-24zM64 472c-13.255 0-24-10.745-24-24s10.745-24 24-24 24 10.745 24 24-10.745 24-24 24zM384 81.452c0 42.416-25.97 66.208-33.277 94.548h101.723c33.397 0 59.397 27.746 59.553 58.098.084 17.938-7.546 37.249-19.439 49.197l-.11.11c9.836 23.337 8.237 56.037-9.308 79.469 8.681 25.895-.069 57.704-16.382 74.757 4.298 17.598 2.244 32.575-6.148 44.632C440.202 511.587 389.616 512 346.839 512l-2.845-.001c-48.287-.017-87.806-17.598-119.56-31.725-15.957-7.099-36.821-15.887-52.651-16.178-6.54-.12-11.783-5.457-11.783-11.998v-213.77c0-3.2 1.282-6.271 3.558-8.521 39.614-39.144 56.648-80.587 89.117-113.111 14.804-14.832 20.188-37.236 25.393-58.902C282.515 39.293 291.817 0 312 0c24 0 72 8 72 81.452z"
              />
            </svg>
            <span
              class="ml-2"
            >
              추천
            </span>
          </label>
          <label
            class="flex justify-center items-center hover:text-basic-red text-sm "
          >
            <input
              class="appearance-none"
              type="radio"
              value="top"
            />
            <svg
              class="text-lg"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 320 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M177 255.7l136 136c9.4 9.4 9.4 24.6 0 33.9l-22.6 22.6c-9.4 9.4-24.6 9.4-33.9 0L160 351.9l-96.4 96.4c-9.4 9.4-24.6 9.4-33.9 0L7 425.7c-9.4-9.4-9.4-24.6 0-33.9l136-136c9.4-9.5 24.6-9.5 34-.1zm-34-192L7 199.7c-9.4 9.4-9.4 24.6 0 33.9l22.6 22.6c9.4 9.4 24.6 9.4 33.9 0l96.4-96.4 96.4 96.4c9.4 9.4 24.6 9.4 33.9 0l22.6-22.6c9.4-9.4 9.4-24.6 0-33.9l-136-136c-9.2-9.4-24.4-9.4-33.8 0z"
              />
            </svg>
            <span
              class="ml-2"
            >
              TOP
            </span>
          </label>
        </div>
        <div
          class="flex items-end w-full h-24 border mb-2 border-basic-gray-third relative text-basic-white "
          style="background-image: url(https://cdn.gameinsight.co.kr/news/photo/202202/24053_61496_3054.jpg); background-repeat: no-repeat; background-size: cover; background-position: 50% 52%;"
        >
          <div
            class="hover-overlay w-full h-full bg-basic-red opacity-0 hover:opacity-100 transition-opacity duration-500 ease-in-out"
            style="background-image: url(https://cdna.artstation.com/p/assets/images/images/030/940/370/small/lorenzo-lanfranconi-west-studio-lorenzo-lanfranconi-valorant-1.jpg?1602102081); background-repeat: no-repeat; background-size: cover; background-position: 50% 45%;"
          />
          <p
            class="absolute bottom-0 right-0 mr-2 mb-1 text-xs "
          >
            홈페이지 바로가기
          </p>
        </div>
        <div>
          <p
            class="text-lg text-center"
          >
            로딩중입니다...
          </p>
        </div>
      </div>
      <div
        class="hidden w-4/12 ml-3 md:block"
      >
        <div
          class="bg-white rounded mb-6"
        >
          <div
            class="p-3 rounded-t bg-basic-black-second"
          >
            <p
              class="text-xl text-basic-white font-ptBlack"
            >
              인기 게시판
            </p>
          </div>
          <div />
        </div>
        <div
          class="bg-basic-white border rounded"
        >
          <div
            class="w-full text-center"
          >
            <a
              class="w-auto p-2 m-4 flex justify-center items-center text-white text-sm bg-basic-red rounded font-ptBlack"
              href="/login"
            >
              로그인
            </a>
          </div>
          <div
            class="w-full text-start pl-4 text-basic-gray-third flex-col pt-2 border-t border-basic-gray-second"
          >
            <p
              class="text-sm text-basic-black-second"
            >
              홈
            </p>
            <a
              class="w-auto m-2 rounded block text-basic-black-second"
              href="/subs/create"
            >
              전체
            </a>
          </div>
          <div
            class="w-full text-start pl-4 text-basic-gray-third flex-col pt-2 border-t border-basic-gray-second"
          >
            <p
              class="text-sm text-basic-black-second"
            >
              정보
            </p>
            <a
              class="w-auto m-2 rounded block text-basic-black-second"
              href="/subs/create"
            >
              요원
            </a>
            <a
              class="w-auto m-2 rounded block text-basic-black-second"
              href="/subs/create"
            >
              팁과 노하우
            </a>
            <a
              class="w-auto m-2 rounded block text-basic-black-second"
              href="/subs/create"
            >
              유저 뉴스
            </a>
          </div>
          <div
            class="w-full text-start pl-4 text-basic-gray-third flex-col pt-2 border-t border-basic-gray-second"
          >
            <p
              class="text-sm text-basic-black-second"
            >
              커뮤니티
            </p>
            <a
              class="w-auto m-2 rounded block text-basic-black-second"
              href="/subs/create"
            >
              자유
            </a>
            <a
              class="w-auto m-2 rounded block text-basic-black-second"
              href="/subs/create"
            >
              유머
            </a>
            <a
              class="w-auto m-2 rounded block text-basic-black-second"
              href="/subs/create"
            >
              영상
            </a>
            <a
              class="w-auto m-2 rounded block text-basic-black-second"
              href="/subs/create"
            >
              유저찾기
            </a>
            <a
              class="w-auto m-2 rounded block text-basic-black-second"
              href="/subs/create"
            >
              사건 신고
            </a>
            <a
              class="w-auto m-2 rounded block text-basic-black-second"
              href="/subs/create"
            >
              팬아트
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
